### PR TITLE
provider/aws: Fix issue with IAM Server Certificates and Chains

### LINKED ISSUE #2713: AWS Server Certificate path argument fails strconv.ParseBool
Terraform v0.6.0

```
variable "l0_prefix" {
    default = "l0"
}

...

resource "aws_iam_server_certificate" "cert" {
    name = "${var.l0_prefix}-cert"
    path = "/layer0/${var.l0_prefix}/"
    certificate_body = "${file("cert/body.crt")}"
    private_key = "${file("cert/key.pem")}"
    certificate_chain = "${file("cert/chain.intermed")}"
}

```

Attempts to run `terraform plan` crash with the following message:
```
aws_iam_server_certificate.cert: cannot parse '' as bool: strconv.ParseBool: parsing "/layer0/l0/": invalid syntax
```

### PR DESCRIPTION
**Pull request by catsby**
Should fix #2713 and #2802 . I don't _think_ it interferes with https://github.com/hashicorp/terraform/pull/2411

### CODE REVIEW

File: builtin/providers/aws/resource_aws_iam_server_certificate.go [hunk 1]
```diff
@@ -132,9 +140,10 @@ func resourceAwsIAMServerCertificateDelete(d *schema.ResourceData, meta interfac
 }
 
 func normalizeCert(cert interface{}) string {
-	if cert == nil {
+	if cert == nil || cert == (*string)(nil) {
```
Comment (phinze) [inline builtin/providers/aws/resource_aws_iam_server_certificate.go]: This doesn't seem like it should be necessary?

https://play.golang.org/p/gED99KcQnP

Comment (catsby) [inline builtin/providers/aws/resource_aws_iam_server_certificate.go]: I didn't think so either, but https://play.golang.org/p/A6vKbXIkfp

Comment (phinze) [inline builtin/providers/aws/resource_aws_iam_server_certificate.go]: Wow... that's unexpected! LGTM

**Comment by phinze** [comment]
LGTM

**Pull request merged**